include/debug_scani2c.h

```diff
@@ -20,24 +21,17 @@ void scanI2C() {
         error = Wire.endTransmission();
 
         if (error == 0) {
-            Serial.print("I2C device found at address 0x");
-            if (address < 16) Serial.print("0");
-            Serial.print(address, HEX);
-            Serial.println("  !");
+            OSW_LOG_I("I2C device found at address 0x", address < 16 ? '0' : '', msg, String(address, HEX), " !");
 
             nDevices++;
         } else if (error == 4) {
-            Serial.print("Unknown error at address 0x");
-            if (address < 16) Serial.print("0");
-            Serial.println(address, HEX);
+            OSW_LOG_E("Unknown error at address 0x", address < 16 ? "0" : "", String(address, HEX));
```

Comment:
Maybe I think it's kind of like this.

>In this case, the conditional statements are always compiled and always passed through the conditional statements, >with or without debug flags. Previously, if statements were compiled only when the NDEBUG flag was declared. Can >this also be included as a function or condition to output the log?
>Use default param??
>```
>OSW_LOG_W("Acceleration API enabled, but no provider available!",!this->accelSensor);
>```
>like this...

Reply:
The interface of the `OswLogger` was explicitly designed to remove the need for string-concatenation. This line is an example how to use it... I don't think putting conditions into potentially never expanded defines is a good idea. Take a look into `assert()` - this function is exactly one of these dangers, as sometimes you put relevant code into its condition, which won't be compiled into release-builds. Very uncool :grin: